src/com/facebook/buck/apple/AppleBundle.java

```diff
@@ -670,6 +672,24 @@ public boolean isLegacyWatchApp() {
     return stepsBuilder.build();
   }
 
+  private void verifyResourceConflicts(AppleBundleResources resources,
+                                       SourcePathResolver resolver) {
+    // Ensure there are no resources that will overwrite each other
+    Set<Path> filenames = new HashSet<>();
+    for (SourcePath path : Iterables.concat(
+        resources.getResourceDirs(),
+        resources.getDirsContainingResourceDirs(),
+        resources.getResourceFiles())) {
+      Path filename = resolver.getRelativePath(path).getFileName();
+      if (filenames.contains(filename)) {
+        throw new HumanReadableException(
+            "Bundle contains multiple resources with filename %s", filename);
```

Comment:
Nit: `filename` can be inaccurate as it can be directories, too. Can we improve the naming?